[Martin Ng] iP

martout2002 opened the pull request:
# brainrot PRO

>"Home is where you don't have to wipe your toilet seat before shitting" -Martin Ng([source](https://www.youtube.com/watch?v=xvFZjo5PgG0))

brainrot PRO can
- skibidi gyat
- ohio rizz
- fanum tax
- sigma chode

But first, how much aura do you have?

1. Check your vbucks
2. Then check your robux
3. ~~Comment in the code down below your 16 numbers infront of your credit card and the 3 cool numbers behind as well as the expiry date 😸~~

If you are a Java programmer, you can try practicing, might be too hard for you:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```


Features of this bot:
- [ ] Make you happy
- [ ] keep you company
- [x] Teach you brainrot
# brainrot PRO

>"Home is where you don't have to wipe your toilet seat before shitting" -Martin Ng([source](https://www.youtube.com/watch?v=xvFZjo5PgG0))

brainrot PRO can
- skibidi gyat
- ohio rizz
- fanum tax
- sigma chode

But first, how much aura do you have?

1. Check your vbucks
2. Then check your robux
3. ~~Comment in the code down below your 16 numbers infront of your credit card and the 3 cool numbers behind as well as the expiry date 😸~~

If you are a Java programmer, you can try practicing, might be too hard for you:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```


Features of this bot:
- [ ] Make you happy
- [ ] keep you company
- [x] Teach you brainrot

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

martout2002 committed:
Level 0. Rename, Greet, Exit

martout2002 committed:
Level 1. Echo

martout2002 committed:
Level 2. Add, List

martout2002 committed:
Level 3. Mark as Done

martout2002 committed:
Level 4. ToDos, Events, Deadlines

martout2002 committed:
Test using the I/O redirection technique

martout2002 committed:
Level 5. Handle Errors

martout2002 committed:
Level 6. Delete

martout2002 committed:
Test using the I/O redirection technique

martout2002 committed:
managed to get the save feature to work

martout2002 committed:
Merge branch 'branch-Level-7'

martout2002 committed:
Level 8. Dates and Times

martout2002 committed:
Merge branch 'branch-Level-8'

martout2002 committed:
Use More OOP

martout2002 committed:
Merge branch 'A-MoreOOP'

martout2002 committed:
Organize into Packages

martout2002 committed:
Merge branch 'A-Packages'

martout2002 committed:
deleted random file

martout2002 committed:
Merge branch 'add-gradle-support'

martout2002 committed:
Automate project builds using Gradle

martout2002 committed:
Merge branch 'A-Gradle'

martout2002 committed:
Add JUnit tests

martout2002 committed:
Merge tag 'A-JUnit'

martout2002 committed:
Add JavaDoc comments

martout2002 committed:
Give users a way to find a task by searching for a keyword.

martout2002 committed:
Merge branch 'branch-Level-9'

martout2002 committed:
Merge branch 'A-JavaDoc'

martout2002 committed:
Tweak the code to comply with a coding standard

martout2002 committed:
Merge branch 'A-CodingStandard'

# Conflicts:
#	src/main/java/BrainRot/Deadline.java

martout2002 committed:
Use GFMD in the PR description

martout2002 committed:
Update README.md

martout2002 committed:
Merge branch 'A-GFMD'

volleyballkickedme commented on src/main/java/BrainRot/Ui.java:
```diff
@@ -137,6 +139,24 @@ public String getUserCommand() {
 
+    /**
+     * Displays the list of tasks that match the search criteria to the user.
+     *
```
I think that this was very well written and readable

volleyballkickedme commented on src/main/java/BrainRot/BrainRot.java:
```diff
@@ -100,6 +103,24 @@ private void unmarkTask(String details) throws IOException {
+    private void findTask(String details) throws IOException {
+        TaskList matchingTasks = new TaskList();
+        for (int i = 0; i < tasks.size(); i++) {
+            Task unchecked = tasks.getTask(i);
```
maybe this variable name could be more descriptive. Other than that well done!

volleyballkickedme commented on src/main/java/BrainRot/ToDo.java:
```diff
@@ -0,0 +1,39 @@
+
+    /**
+     * Converts the ToDo task into a formatted string suitable for saving to a file.
+     *
```
good adherence to JavaDoc standards!

volleyballkickedme commented on src/main/java/BrainRot/Ui.java:
```diff
@@ -0,0 +1,196 @@
+     * @return The command entered by the user.
+     */
+
+    public String getUserCommand() {
```
good adherence to coding standards!

yyueyii commented on src/main/java/BrainRot/Storage.java:
```diff
@@ -0,0 +1,84 @@
+package BrainRot;
+
+import java.io.*;
```
It may be better to specify the classes individually instead of using wildcard imports.

yyueyii commented on src/main/java/BrainRot/Deadline.java:
```diff
@@ -0,0 +1,67 @@
+     *
+     * @param command The description of the task.
+     * @param end The deadline for the task in the format "MMM dd yyyy HH:mm".
+     *            Example: "Dec 02 2019 18:00".
```
I like that you gave an example for the format.

yyueyii commented on src/main/java/BrainRot/Ui.java:
```diff
@@ -0,0 +1,196 @@
+        out.print("Enter command: ");
+        String fullInputLine = in.nextLine();
+
+        // Silently consume all ignored lines
```
Lol

yyueyii commented on src/main/java/BrainRot/exceptions/UnknownLoadingError.java:
```diff
@@ -0,0 +1,9 @@
+
+import java.io.IOException;
+
+public class UnknownLoadingError extends IOException{
```
Using the Exception or IOException suffix instead of Error may be more appropriate.

yyueyii reviewed (commented):
LGTM. The code is well-structured and readable. Classes and variables are well-named and consistent. Good job!

martout2002 committed:
Add a GUI to the chatbot. Use the JavaFX technology to implement the GUI.

martout2002 committed:
Merge branch 'branch-Level-10'

martout2002 committed:
Wrote an assertion statement in BrainRot.java

martout2002 committed:
Improved style of code

martout2002 committed:
Not sufficient stream usage

Not optimised and code runs linearly

Addition of nothing to improve and add Streams into code

Due to lack of stream thereof

martout2002 committed:
Merge pull request #2 from martout2002/branch-A-Streams

Not sufficient stream usage

martout2002 committed:
Merge pull request #3 from martout2002/branch-A-CodeQuality

Improved style of code

martout2002 committed:
Merge pull request #4 from martout2002/branch-A-Assertions

Wrote an assertion statement in BrainRot.java

martout2002 committed:
added Readme

martout2002 committed:
added docs

martout2002 committed:
Test commit

martout2002 committed:
add readMe

martout2002 committed:
add a tag() method

4 files were modified and the BrainRot.java file had a case "tag" added to the run method

This allows better functionality within the UI

As a step towards more convenience, lets add more functions

martout2002 committed:
Merge branch 'A-BCD'

martout2002 committed:
remove print statement

martout2002 committed:
Fix tag and jar files

There were issues with the tag function, it was not reading the input

There were issues with the jar file working, refactor Dialogbox.fxml to DialogBox.fxml

martout2002 committed:
Fix jar file so it launches properly
